MaterializeInc/materialize
# sql-parser: parse kafka consistency two ways

### Motivation

Our `CREATE SINK` syntax has [a bug](https://github.com/MaterializeInc/materialize/issues/8231). This PR updates the parser to parse the corrected `CREATE SINK` syntax, without removing the broken one. The idea, which was discussed today in the SQL standup and [in Slack](https://materializeinc.slack.com/archives/CMHDK0DK8/p1632242364299100), is that we will deprecate the broken syntax in X versions. By not removing the broken syntax, we are avoiding a migration.



### Description

This PR parses an updated `CONSISTENCY` clause, outlined by the following diagrams.

<img width="760" alt="Screen Shot 2021-09-21 at 4 13 09 PM" src="https://user-images.githubusercontent.com/5766027/134259672-015da989-67e3-428a-9f1b-497f01bc81c8.png">
<img width="760" alt="Screen Shot 2021-09-21 at 4 13 22 PM" src="https://user-images.githubusercontent.com/5766027/134259687-0c2711a5-f413-46d6-b196-736e803b2973.png">
<img width="760" alt="Screen Shot 2021-09-21 at 4 13 38 PM" src="https://user-images.githubusercontent.com/5766027/134259693-f455daf6-ab39-4fb6-a7df-4b4f129be7bf.png">



### Manual testing

I manually tested this change (without the `testdrive` magic) to ensure that the Kafka topics created for sinks were correctly nonced/not nonced.

The old syntax continues to cause bugs. The updated syntax will not nonce the created Kafka topic if `reuse_topic` is `true`, regardless of the consistency `FORMAT` clause.



### Checklist

- [X] This PR has adequate test coverage.
- [X] This PR adds a release note for any user-facing behavior changes.
